yaksh/views.py

```diff
@@ -1025,3 +1032,37 @@ def edit_profile(request):
         context['form'] = form
         return my_render_to_response('yaksh/editprofile.html', context,
                                     context_instance=ci)
+
+
+@login_required
+def download_questions(request):
+    user = request.user
+    if not is_moderator(user):
+        raise Http404('You are not allowed to view this page!')
+    question = Question()
+    questions = question.dump_into_json(user)
+    response = HttpResponse(questions, content_type='text/json')
+    response['Content-Disposition'] = 'attachment; filename="{0}_questions.json"'\
+                                                            .format(user)
+    return response
+
+
+@login_required
+def upload_questions(request):
+    user = request.user
+    ci = RequestContext(request)
+
+    if not is_moderator(user):
+        raise Http404('You are not allowed to view this page!')
+    if request.method == 'POST':
+        form = UploadFileForm(request.POST, request.FILES)
+        if form.is_valid():
+            questions_file = request.FILES['file']
+            if questions_file.name.split('.')[1] == "json":
+                questions_list = questions_file.read()
+                question = Question()
+                question.load_from_json(questions_list, user)
+                return my_redirect('/exam/manage/questions')
+            else:
+                raise Http404("Please Upload a JSON file")
```

Comment:
Is it more suitable to keep the user on the same page and print a Warning in a notification bar saying ""Please Upload a JSON file""
Example (for representational purposes):
![](http://i.stack.imgur.com/iMLdo.png)